### PR TITLE
Enable fallback on Docker config using custom auth file

### PR DESCRIPTION
**Pull request by jdolitsky**
related to https://github.com/helm/helm/pull/10536
cc @scottrigby

### CODE REVIEW

**Review by sajayantony — APPROVED** [review]

**Comment by sajayantony** [comment]
@jdolitsky @Wwwsylvia should this be in backported to main or has main removed need for docker-cli and it way more complicated now?

File: pkg/auth/docker/client.go [hunk 1]
```diff
@@ -64,6 +64,39 @@ func NewClient(configPaths ...string) (auth.Client, error) {
 	}, nil
 }
 
+// NewClientWithDockerFallback creates a new auth client
+// which falls back on Docker's default config path.
+// This allows support for ~/.docker/config.json as a fallback,
+// as well as support for the DOCKER_CONFIG environment variable.
+func NewClientWithDockerFallback(configPaths ...string) (auth.Client, error) {
+	if len(configPaths) == 0 {
+		return nil, errors.New("must provide at least one config path")
```
Comment (sabre1041) [inline pkg/auth/docker/client.go:73]: Would we want to call `NewClient` when no config paths have been provided? Or is this a way to enforce the end user call `NewClient` if they do not provide any `configPaths`

Comment (jdolitsky) [reply pkg/auth/docker/client.go]: Yes, done

**Comment by shizhMSFT** [comment]
> should this be in backported to main or has main removed need for docker-cli and it way more complicated now?

This should not be backported to `main` as `main` will remove `pkg/` eventually.

**Pull request merged**